include/stl2/detail/algorithm/partial_sort_copy.hpp

```diff
@@ -28,57 +28,57 @@
 // partial_sort_copy [partial.sort.copy]
 //
 STL2_OPEN_NAMESPACE {
-	template<InputIterator I1, Sentinel<I1> S1,
-		RandomAccessIterator I2, Sentinel<I2> S2,
-		class Comp = less,
-		class Proj1 = identity, class Proj2 = identity>
-	requires
-		IndirectlyCopyable<I1, I2> &&
-		Sortable<I2, Comp, Proj2> &&
-		IndirectStrictWeakOrder<
-			Comp, projected<I1, Proj1>, projected<I2, Proj2>>
-	I2 partial_sort_copy(I1 first, S1 last, I2 result_first, S2 result_last,
-		Comp comp = {}, Proj1 proj1 = {}, Proj2 proj2 = {})
-	{
-		auto r = result_first;
-		if(r != result_last) {
-			auto cresult = ext::copy(std::move(first), last, std::move(r), result_last);
-			first = std::move(cresult.in);
-			r = std::move(cresult.out);
+	struct __partial_sort_copy_fn : private __niebloid {
+		template<InputIterator I1, Sentinel<I1> S1, RandomAccessIterator I2, Sentinel<I2> S2,
+			class Comp = less, class Proj1 = identity, class Proj2 = identity>
+		requires IndirectlyCopyable<I1, I2> &&
+			Sortable<I2, Comp, Proj2> &&
+			IndirectStrictWeakOrder<Comp, projected<I1, Proj1>, projected<I2, Proj2>>
```

Comment:
Why not use the *constrained-parameter* form for `Comp`?

Reply:
This should be made upstream too.
https://timsong-cpp.github.io/cppwp/partial.sort.copy